File: go/hamming/hamming.go
package hamming

import "errors"

const testVersion = 6

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("Input strings must be equal length")
	}
	distance := 0
	for i := range a {
		if a[i] != b[i] {
			distance++
		}
	}
	return distance, nil
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
/**
 * Created by zhangchf on 28/08/2017.
 */
package Hamming

fun compute(leftStrand: String, rightStand: String): Int {
    require(leftStrand.length == rightStand.length) {"leftStrand and rightStrand must be of equal length."}

    return leftStrand.zip(rightStand).count { (left, right) -> left != right }
}
